dvc/tree/s3.py

```diff
@@ -54,6 +54,9 @@ def __init__(self, repo, config):
 
         self._append_aws_grants_to_extra_args(config)
 
+        self.access_key_id = config.get("access_key_id")
+        self.secret_access_key = config.get("secret_access_key")
```

Comment:
Why do we assign it instead of using `self.config` directly later?

Reply:
Following the pattern in other remotes (OSS, GDrive etc).